src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+        num_images = len(input_image)
+
+        # latent checks
+        input_latent_stacked = False
+        if input_latent is not None:
+            if torch.is_tensor(input_latent):
+                if input_latent.ndim == 5:
+                    input_latent = [input_latent[i] for i in range(input_latent.shape[0])]
+                    input_latent_stacked = True
+                elif input_latent.ndim == 4:
+                    input_latent = [input_latent]
+                else:
+                    raise ValueError(f"Unsupported number of dimension in the input latent: {input_latent.ndim}.")
+            if isinstance(input_latent, list) and not isinstance(input_latent, tuple):
+                if not all(torch.is_tensor(k) for k in input_latent):
+                    raise ValueError("Input latent must be a torch.FloatTensor.")
+                if not all(
+                    k.dim() == 4 and k.shape[0] == ensemble_size and k.shape[1] == self.latent_space_size
+                    for k in input_latent
+                ):
+                    raise ValueError(
+                        f"Input latent must be 4-dimensional with shape [E,{self.latent_space_size},h,w], "
+                        f"where E is the requested ensemble_size."
+                    )
+                if len(input_latent) != num_images:
+                    raise ValueError(
+                        f"The numbers of input images ({num_images}) and latents ({len(input_latent)}) "
+                        f"are not compatible."
+                    )
+            else:
+                raise ValueError(f"Unsupported latent type: {type(input_latent)}.")
+        if input_image_stacked ^ input_latent_stacked:
+            logger.warning("Different stacking of input images and latents might be a sign of undesired behavior.")
+
+        if generator is not None:
+            if isinstance(generator, torch.Generator):
+                generator = [generator]
+            if isinstance(generator, list) and not isinstance(generator, tuple):
+                if len(generator) == 1 and num_images > 1:
+                    generator = generator * num_images
+                if len(generator) != num_images:
+                    raise ValueError(
+                        f"The numbers of input images ({num_images}) and generators ({len(generator)}) "
+                        f"are not compatible."
+                    )
+            else:
+                raise ValueError(f"Unsupported generator type: {type(generator)}.")
+
+        # Prepare the empty text embedding. In the future, remove text modules completely
+        if self.empty_text_embedding is None:
+            self.encode_empty_text()
+
+        out = []
+        with self.progress_bar(total=num_images * ensemble_size * denoising_steps) as progress_bar:
+            for i in range(num_images):
+                out.append(
+                    self.process_image(
```

Comment:
can we unwrap this function? 
we should not wrap almost all of the logic inside a different function in our denosing loop. this is not consistent with our pipeline design and IMO make the code much less readable. If you read our pipelines, you can see that we only abstract away details with methods such as `prepare_latents`, and `encode_prompts`; and we make sure the definitions of these methods are pretty consistent across pipelines

Reply:
can we roughtly follow this structure?

```python

# this include load_image_canonical, resize, pad_image
image = preprocess_image(input_image)

# here we will include encode_image + prepare_latent 
latents = self.prepare_latent(image, input_latent, generator, ensemble_size)

predictions = []

for i in range(0, ensemble_size, batch_size):
    i_end = min(i + batch_size, ensemble_size)
    batch_latents = latents[i:i_end]  # [B,4,h,w]
    timesteps = self.scheduler.set_timesteps(denoising_steps)
    for i, t in enumerate(timesteps):
        ...
        pred_noise = self.unet(...)
        ...
        batch_latents = self.scheduler.step(...)
       # here we decode each batch latent in the same loop and get a prediction directly
        prediction = self.decode_image(batch_latents)
    ....
    predictions.append(prediction) 

predictions = torch.cat(predictions, dim=0)
...

# ensemble_depth etc
...
```

Reply:
Fixed